compendium/DeclarativeServices/test/TestUpdateConfiguration.cpp

```diff
@@ -0,0 +1,117 @@
+   */
+TEST_F(tServiceComponent, testUpdateConfig_Modified) //DS_CAI_FTC_1
+{
+  //start bundle
+  cppmicroservices::Bundle testBundle = StartTestBundle("TestBundleDSCAI1");
+
+  // Use DS runtime service to validate the component description
+  scr::dto::ComponentDescriptionDTO compDescDTO =
+    dsRuntimeService->GetComponentDescriptionDTO(testBundle,
+                                                 "sample::TestBundleDSCAI1");
+  EXPECT_EQ(compDescDTO.implementationClass, "sample::TestBundleDSCAI1")
+    << "Implementation class in the returned component description must be "
+       "sample::TestBundleDSCAI1";
+
+  // Use DS runtime service to validate the component state.
+  //It should be in the SATISFIED state because the configuration policy is optional.
+  //and component is delayed
+  auto compConfigs =
+    dsRuntimeService->GetComponentConfigurationDTOs(compDescDTO);
+  EXPECT_EQ(compConfigs.size(), 1ul) << "One default config expected";
+  EXPECT_EQ(compConfigs.at(0).state, scr::dto::ComponentState::SATISFIED);
+
+  //getservice to make component active
+  auto ctxt = framework.GetBundleContext();
+  auto sRef1 = ctxt.GetServiceReference<test::Interface4>();
+  EXPECT_TRUE(static_cast<bool>(sRef1)) << "Service must be available";
+  auto service = ctxt.GetService<test::Interface4>(sRef1);
+  ASSERT_NE(service, nullptr);
+
+  compConfigs = dsRuntimeService->GetComponentConfigurationDTOs(compDescDTO);
+  EXPECT_EQ(compConfigs.size(), 1ul) << "One default config expected";
+  EXPECT_EQ(compConfigs.at(0).state, scr::dto::ComponentState::ACTIVE);
+
+  // Get a service reference to ConfigAdmin to create the component instance.
+  auto caRef =
+    framework.GetBundleContext()
+      .GetServiceReference<cppmicroservices::service::cm::ConfigurationAdmin>();
+  ASSERT_TRUE(caRef) << "GetServiceReference failed for ConfigurationAdmin";
+  auto configAdminService =
+    framework.GetBundleContext()
+      .GetService<cppmicroservices::service::cm::ConfigurationAdmin>(caRef);
+  ASSERT_TRUE(configAdminService) << "GetService failed for ConfigurationAdmin";
+
+  //Create configuration object
+  auto configObject =
+    configAdminService->GetConfiguration("sample::TestBundleDSCAI1");
+  auto configObjInstance = configObject->GetPid();
+
+  //Update property
+  cppmicroservices::AnyMap props(
+    cppmicroservices::AnyMap::UNORDERED_MAP_CASEINSENSITIVE_KEYS);
+  const std::string instanceId{ "instance1" };
+  props["uniqueProp"] = instanceId;
+  configObject->Update(props);
+
+  auto result = RepeatTaskUntilOrTimeout(
+    [&compDescDTO, &compConfigs, this, &testBundle, &configObjInstance]() {
+      compDescDTO = dsRuntimeService->GetComponentDescriptionDTO(
+        testBundle, configObjInstance);
+      if (compDescDTO.name != "") {
+        compConfigs =
+          this->dsRuntimeService->GetComponentConfigurationDTOs(compDescDTO);
+      }
+    },
+    [&compConfigs, &instanceId]() -> bool {
+      if (compConfigs.size() == 1) {
```

Comment:
This logic confirms that DS updated its local copy of the properties. DS updates its local copy  before it calls the Modified method for the component instance though so this doesn't verify that the component instance's Modified method was called. Should add a GetProperties method to the interface and call it to confirm that the Modified method of the component instance was called and the properties updated appropriately or use test::CAInterface. That one already has a GetProperties method.